lib/ansible/modules/cloud/azure/azure_rm_eventhubnamespace.py

```diff
@@ -0,0 +1,318 @@
+            self.location = resource_group.location
+        self.sku = str.capitalize(self.sku) if self.sku else None
+        eventhubnamespace = self.get_namespace() or dict()
+        if self.state == 'present':
+            if not eventhubnamespace:
+                changed = True
+                if self.kafka_enabled and self.sku == 'Basic':
+                    self.module.warn('kafka_enabled cannot be set to true under basic sku and would be ignored')
+                    self.kafka_enabled = None
+
+                if self.sku:
+                    self.sku = self.eventhub_models.Sku(name=self.sku, tier=self.sku)
+
+                eventhubnamespace_instance = self.eventhub_models.EHNamespace(location=self.location,
+                                                                              sku=self.sku,
+                                                                              tags=self.tags,
+                                                                              is_auto_inflate_enabled=self.auto_inflate_enabled,
+                                                                              maximum_throughput_units=self.maximum_throughput_units,
+                                                                              kafka_enabled=self.kafka_enabled)
+                if not self.check_mode:
+                    eventhubnamespace = self.create_or_update_namespace(eventhubnamespace_instance)
+            else:
+                changed, eventhubnamespace = self.check_status(changed=changed, eventhubnamespace=eventhubnamespace)
+
+                if changed and not self.check_mode:
+                    eventhubnamespace = self.create_or_update_namespace(eventhubnamespace)
+            results = self.to_dict(eventhubnamespace)['id'] if eventhubnamespace else None
+        elif eventhubnamespace:
+            changed = True
+            if not self.check_mode:
+                self.delete_namespace()
+
+        self.results['id'] = results
+        self.results['changed'] = changed
+        return self.results
+
+    def create_or_update_namespace(self, eventhubnamespace):
+        try:
+            self.log('Creating or updating an event hub namespace')
+            poller = self.eventhub_client.namespaces.create_or_update(resource_group_name=self.resource_group,
+                                                                      namespace_name=self.name,
+                                                                      parameters=eventhubnamespace)
+            return self.get_poller_result(poller)
+        except self.eventhub_models.ErrorResponseException as exc:
+            self.fail('Error creating or updating Event Hub Namespace {0}: {1}'.format(self.name, str(exc.inner_exception) or str(exc.message) or str(exc)))
+
+    def check_status(self, changed, eventhubnamespace):
+        # Compare sku
+        if self.sku and self.sku != eventhubnamespace.sku.name:
```

Comment:
default value of sku is error-prone. suppose first time user set sku as premium or something, second time not setting sku, so default is basic, premium!=basic, but actually user don't want to change this field.

Reply:
I checked the code. During each update process, if the sku is not set, the current sku will be wrote back explicitly. I deleted the confusing part of the documentation. Thanks!